pkg/operator/operator.go

```diff
@@ -133,51 +139,45 @@ func (c *vSphereProblemDetectorController) sync(ctx context.Context, syncCtx fac
 		return err
 	}
 
-	availableCnd := operatorapi.OperatorCondition{
-		Type:   controllerName + operatorapi.OperatorStatusTypeAvailable,
-		Status: operatorapi.ConditionTrue,
+	if !platformSupported {
+		return nil
 	}
 
-	blockUpgrade := false
-	var blockUpgradeReason string
+	clusterInfo := util.NewClusterInfo()
+	delay, checkPerformed := c.runSyncChecks(ctx, clusterInfo)
 
-	// TODO: Run in a separate goroutine? We may not want to run time-consuming checks here.
-	if platformSupported && time.Now().After(c.nextCheck) {
-		clusterInfo := util.NewClusterInfo()
-		delay, err := c.runChecks(ctx, clusterInfo)
-		if err != nil {
-			klog.Errorf("Failed to run checks: %s", err)
-		}
-		// Do not return the error, it would degrade the whole cluster.
-		// Remember the error and put it in Available condition message below.
-		c.lastError = err
-
-		blockUpgrade, blockUpgradeReason = c.checkForDeprecation(clusterInfo)
-		// if we blocked upgrades then we should retry sooner
-		if blockUpgrade {
-			delay = c.backoff.Step()
-		}
+	syncErrorValue := 0
+	if c.lastClusterCheckResult.checkError != nil {
+		syncErrorValue = 1
+	}
+	syncErrrorMetric.WithLabelValues("SyncError").Set(float64(syncErrorValue))
 
-		// Poke the controller sync loop after the delay to re-run tests
-		queue := syncCtx.Queue()
-		queueKey := syncCtx.QueueKey()
-		c.nextCheck = c.lastCheck.Add(delay)
-		klog.V(2).Infof("Scheduled the next check in %s (%s)", delay, c.nextCheck)
-		time.AfterFunc(delay, func() {
-			queue.Add(queueKey)
-		})
+	// if no checks were performed don't update conditons
+	if !checkPerformed {
+		return nil
+	}
```

Comment:
When checks were not run, we simply return. I tested this and this would prevent flip-flop of conditions when checks are not run